ql/src/java/org/apache/hadoop/hive/ql/optimizer/GlobalLimitOptimizer.java

```diff
@@ -99,27 +99,31 @@ public ParseContext transform(ParseContext pctx) throws SemanticException {
         LimitDesc tempGlobalLimitDesc = tempGlobalLimit.getConf();
         Table tab = ts.getConf().getTableMetadata();
         Set<FilterOperator> filterOps = OperatorUtils.findOperators(ts, FilterOperator.class);
-
-        if (!tab.isPartitioned()) {
-          if (filterOps.size() == 0) {
-            Integer tempOffset = tempGlobalLimitDesc.getOffset();
-            globalLimitCtx.enableOpt(tempGlobalLimitDesc.getLimit(),
-                (tempOffset == null) ? 0 : tempOffset);
-          }
-        } else {
-          // check if the pruner only contains partition columns
-          if (onlyContainsPartnCols(tab, filterOps)) {
-
-            String alias = (String) topOps.keySet().toArray()[0];
-            PrunedPartitionList partsList = pctx.getPrunedPartitions(alias, ts);
-
-            // If there is any unknown partition, create a map-reduce job for
-            // the filter to prune correctly
-            if (!partsList.hasUnknownPartitions()) {
+        // StorageHandlers will always have empty tablePath.
+        // GenMapRedUtils.setMapWork removes empty tablePath from input dir with select-Limit
+        // InputFormat.getSplits wont be called if no input path & TS Vertex will have 0 task parallelism
+        if (tab.getStorageHandler() == null) {
```

Comment:
Can we move this check to just after getting `TableScanOperator` i.e. line 94/95? We will avoid un-necessary `checkQBpForGlobalLimit` and `findOperators`.

Reply:
Handled in the new commit Vineet. Thanks for review.